customcommands/tmplextensions.go

```diff
@@ -488,6 +497,118 @@ func tmplDBDelById(ctx *templates.Context) interface{} {
 		return "", err
 	}
 }
+func tmplDBDelMultiple(ctx *templates.Context) interface{} {
+	return func(query interface{}, iAmount interface{}, iSkip interface{}) (interface{}, error) {
+		if ctx.IncreaseCheckCallCounterPremium("db_interactions", 10, 50) {
+			return "", templates.ErrTooManyCalls
+		}
+
+		if ctx.IncreaseCheckCallCounterPremium("db_multiple", 2, 10) {
+			return "", templates.ErrTooManyCalls
+		}
+
+		ctx.GS.UserCacheDel(CacheKeyDBLimits)
+
+		dict, err := templates.StringKeyDictionary(query)
+		if err != nil {
+			return "", err
+		}
+		var q Query
+		encoded, err := json.Marshal(dict)
+		if err != nil {
+			return "", err
+		}
+		err = json.Unmarshal(encoded, &q)
+		if err != nil {
+			return "", err
+		}
+
+		amount := int(templates.ToInt64(iAmount))
+		if amount > 100 {
+			amount = 100
+		}
+		skip := int(templates.ToInt64(iSkip))
+		orderby := "value_num DESC, id DESC"
+		if q.Reverse {
+			orderby = "value_num ASC, id ASC"
+		}
+
+		qms := []qm.QueryMod{qm.Where("guild_id = ?", ctx.GS.ID), qm.OrderBy(orderby), qm.Limit(amount), qm.Offset(skip)}
+		if q.Pattern.Valid {
+			qms = append(qms, qm.Where("key LIKE ?", limitString(q.Pattern.String, 256)))
+		}
+		if q.UserID.Valid {
+			qms = append(qms, qm.Where("user_id = ?", q.UserID.Int64))
+		}
+		rows, err := models.TemplatesUserDatabases(qms...).AllG(context.Background())
+		if err != nil {
+			return "", err
+		}
+		cleared, err := rows.DeleteAllG(context.Background())
+
+		return cleared, err
+	}
+}
+
+func tmplDBRank(ctx *templates.Context) interface{} {
+	return func(query interface{}, userID int64, key string) (interface{}, error) {
+		if ctx.IncreaseCheckCallCounterPremium("db_interactions", 10, 50) {
+			return "", templates.ErrTooManyCalls
+		}
+
+		if ctx.IncreaseCheckCallCounterPremium("db_multiple", 2, 10) {
+			return "", templates.ErrTooManyCalls
+		}
+
+		dict, err := templates.StringKeyDictionary(query)
+		if err != nil {
+			return "", err
+		}
+
+		var q Query
+		encoded, err := json.Marshal(dict)
```

Comment:
move the query had decoding and whatnot stuff into its own function, like "queryFromDict" or something

Reply:
made a queryFromArg function